aten/src/ATen/native/mps/operations/RangeFactories.mm

```diff
@@ -53,8 +53,57 @@
     }
     bool is_contiguous = result.is_contiguous();
     Tensor r = !is_contiguous ? at::empty_like(result, LEGACY_CONTIGUOUS_MEMORY_FORMAT) : result;
+    using namespace mps;
+    struct CachedGraph : public MPSCachedGraph {
+      CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+      MPSGraphTensor *startTensor = nil;
+      MPSGraphTensor *multiplyTensor = nil;
+      MPSGraphTensor *outputTensor = nil;
+    };
+    auto cache_ = MPSGraphCache::getInstance();
+    auto stream = getCurrentMPSStream();
+    auto mpsDataType = getMPSDataType(result.scalar_type());
+    @autoreleasepool {
+      string key = "arange_mps_out:" + getTensorsStringKey({result}) + ":" + to_string(size);
+      auto cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(key));
+      if (!cachedGraph) {
+        auto *tmpCachedGraph = cache_->CreateCachedGraph(key, ^ MPSCachedGraph *() {
+          CachedGraph *newCachedGraph = nil;
+          @autoreleasepool {
+            auto mpsGraph = make_mps_graph();
+            newCachedGraph = new CachedGraph(mpsGraph);
+            auto shapeTensor = [mpsGraph constantWithData:[NSData dataWithBytes:&size length:sizeof(size)]
+                                                    shape: @[@1]
+                                                 dataType:MPSDataTypeInt64];
+            auto coordsTensor = [mpsGraph coordinateAlongAxis:0
+                                              withShapeTensor:shapeTensor
+                                                         name:nil];
+            coordsTensor = [mpsGraph castTensor:coordsTensor toType:mpsDataType name:@"coords"];
+
+            newCachedGraph->startTensor = mpsGraphRankedPlaceHolder(mpsGraph, mpsDataType, @[@1]);
+            newCachedGraph->multiplyTensor = mpsGraphRankedPlaceHolder(mpsGraph, mpsDataType, @[@1]);
+            auto scaledCoords = [mpsGraph multiplicationWithPrimaryTensor:coordsTensor
+                                                          secondaryTensor:newCachedGraph->multiplyTensor
+                                                                     name:nil];
+            newCachedGraph->outputTensor = [mpsGraph additionWithPrimaryTensor:scaledCoords
+                                                               secondaryTensor:newCachedGraph->startTensor
```

Comment:
Without the `clamp` operation the rest of the operations are similar to linspace. Can we please refactor this so that we generate the same graph for both and add clamp in the end for linspace. That will automatically take care of the above comment.